airbyte-lib/airbyte_lib/factories/_cache_factories.py

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+
+
+import ulid
+from airbyte_lib.caches.duckdb import DuckDBCache, DuckDBCacheConfig
+from airbyte_protocol.models import ConfiguredAirbyteCatalog
+
+
+def get_default_cache() -> DuckDBCache:
+    """Get a local cache for storing data, using the default database path.
+
+    Cache files are stored in the `.cache` directory, relative to the current
+    working directory.
+    """
+    config = DuckDBCacheConfig(
+        db_path="./.cache/default_cache_db.duckdb",
+    )
+    return DuckDBCache(config=config)
+
+
+def new_local_cache(
+    cache_name: str | None = None,
```

Comment:
should we just make this the full path so users can store their duckdb file wherever?

Reply:
My thought was that this would just be used for convenience - where you care only about controlling "reuse" and "side effects" between invocations over multiple executions, but you don't care as much about where/how the data is stored. 

That's the main reason duckdb isn't mentioned here - trying to remove friction. The "name" establishes a unique ID for the cache for purposes of having confidence that you are writing to a new or existing cache - but doesn't expose more internals than that. (For instance, the user doesn't have to worry about what file extension to use or what directories are writeable and git ignored.)

For full control, I think users would need to directly create a DuckDBCache object instead of using this factory function. I'm happy to revisit this discussion though in follow-ons. Probably helpful to think through if we want to use Pydantic for config classes of Snowflake, Postgres, BigQuery, etc. Then, how we treat the "full control" DuckDB cache can follow the same pattern.

Reply:
I think just to give a balance and not be too magical, I'll add an optional arg to override the base directory. This, I think, gives a nice added layer of (optional) control while not requiring an understanding of DuckDB. And being optional will allow AirbyteLib to give a sensible default so the user doesn't need to go spelunking (for instance, especially in notebooks or other runtimes where they might not be familiar with the local filesystem).

Reply:
Done. A new optional "root_dir" arg now allows the base directory to be overridden.